crates/manifest/src/compat.rs

```diff
@@ -120,7 +120,11 @@ pub(crate) fn convert_allowed_http_to_allowed_hosts(
         Vec::new()
     };
     match http_hosts {
-        AllowedHttpHosts::AllowAll => outbound_hosts.extend(["https://*:*".into()]),
+        AllowedHttpHosts::AllowAll => outbound_hosts.extend([
+            "http://*:*".into(),
+            "https://*:*".into(),
+            "http://self".into(),
```

Comment:
Does this also need `https://self` or is `http://self` a special case that allows both?

Reply:
It's not that `http://self` is special, but when a request is made to `self` we check that there is either permission for `http://self` or `https://self`.